static/app/components/events/eventEntries.tsx

```diff
@@ -70,276 +65,296 @@ type Props = {
   api: Client;
   event?: Event;
   group?: Group;
+  isShare?: boolean;
+  showExampleCommit?: boolean;
+  showTagSummary?: boolean;
+  isBorderless?: boolean;
   className?: string;
-} & typeof defaultProps;
-
-type State = {
-  isLoading: boolean;
-  proGuardErrors: ProGuardErrors;
 };
 
-class EventEntries extends Component<Props, State> {
-  static defaultProps = defaultProps;
-
-  state: State = {
-    isLoading: true,
-    proGuardErrors: [],
-  };
+const EventEntries = memo(
+  ({
+    organization,
+    project,
+    location,
+    api,
+    event,
+    group,
+    className,
+    isShare = false,
+    showExampleCommit = false,
+    showTagSummary = true,
+    isBorderless = false,
+  }: Props) => {
+    const [isLoading, setIsLoading] = useState(true);
+    const [proGuardErrors, setProGuardErrors] = useState<ProGuardErrors>([]);
+    const [attachments, setAttachments] = useState<EventAttachment[]>([]);
+
+    const orgSlug = organization.slug;
+    const projectSlug = project.slug;
+    const orgFeatures = organization?.features ?? [];
+
+    const hasEventAttachmentsFeature = orgFeatures.includes('event-attachments');
+
+    useEffect(() => {
+      checkProGuardError();
+      recordIssueError();
+      fetchAttachments();
+    }, []);
+
+    function recordIssueError() {
+      if (!event || !event.errors || !(event.errors.length > 0)) {
+        return;
+      }
 
-  componentDidMount() {
-    this.checkProGuardError();
-    this.recordIssueError();
-  }
+      const errors = event.errors;
+      const errorTypes = errors.map(errorEntries => errorEntries.type);
+      const errorMessages = errors.map(errorEntries => errorEntries.message);
 
-  shouldComponentUpdate(nextProps: Props, nextState: State) {
```

Comment:
Would some of these conditional checks need to be moved to `React.memo()`'s 2nd argument?

Reply:
I did some checks and that wasn't needed

Reply:
I'll double-check everything on staging  😉